File: scripts/debug/probe_pair.js
const { ethers } = require("hardhat");

async function main() {
  const provider = new ethers.JsonRpcProvider("https://zeus-rpc.mainnet.kortana.xyz");
  const pairAddr = "0x49B25560B081d2912b5eB08062e01B42692A539a";

  const pair = new ethers.Contract(pairAddr, [
    "function getReserves() view returns (uint256, uint256, uint32)",
    "function totalSupply() view returns (uint256)"
  ], provider);

  const [reserve0, reserve1] = await pair.getReserves();
  const totalSupply = await pair.totalSupply();
  
  console.log("Reserves:", reserve0.toString(), reserve1.toString());
  console.log("TotalSupply:", totalSupply.toString());
}

main().catch(console.error);

File: scripts/simulation/probe_pair.py
import time
from web3 import Web3

# [INSTITUTIONAL REGISTRY]
RPC_URL = "https://zeus-rpc.mainnet.kortana.xyz"
ROUTER_ADDRESS = "0xF31bB887853F7092706D7E2Cd9c9A7842f03Ca33"
PAIR_ADDRESS = "0x85782446B5ac7c4BcCE639698042e3Ebf46d5242"

PAIR_ABI = [
    {"constant": True, "inputs": [], "name": "getReserves", "outputs": [{"name": "_reserve0", "type": "uint112"}, {"name": "_reserve1", "type": "uint112"}, {"name": "_blockTimestampLast", "type": "uint32"}], "type": "function"},
    {"constant": True, "inputs": [], "name": "token0", "outputs": [{"name": "", "type": "address"}], "type": "function"},
    {"constant": True, "inputs": [], "name": "token1", "outputs": [{"name": "", "type": "address"}], "type": "function"},
]

w3 = Web3(Web3.HTTPProvider(RPC_URL))

def probe_reserves():
    print(f"[PROBE] TARGET PAIR: {PAIR_ADDRESS}")
    pair = w3.eth.contract(address=w3.to_checksum_address(PAIR_ADDRESS), abi=PAIR_ABI)
    
    try:
        t0 = pair.functions.token0().call()
        t1 = pair.functions.token1().call()
        reserves = pair.functions.getReserves().call()
        
        print(f"[DATA] Token 0: {t0}")
        print(f"[DATA] Token 1: {t1}")
        print(f"[DATA] Reserve 0: {reserves[0] / 1e18}")
        print(f"[DATA] Reserve 1: {reserves[1] / 1e18}")
        
    except Exception as e:
        print(f"[ERROR] Pair probe failed: {e}")

if __name__ == "__main__":
    probe_reserves()
